### PR TITLE
Conda activate fix and development environment

### LINKED ISSUE #651: [BUG] Bugs in Conda activation command set up
**Describe the bug**
The current `install_shapepipe` script includes some bugs that cause issues for creating the `shapepipe` environment with some versions of Conda.

This issue stems from the following line:

```bash
  if [ "$CONDA_VERSION_MINOR" -gt 4 ]
```

which checks the minor version of Conda. This will continuously cause issues every time there is a new major release of Conda.

**To Reproduce**
```bash
./install_shapepipe
```

**Expected behaviour**
The `shapepipe` environment should be created.

### PR DESCRIPTION
**Pull request by sfarrens**
## Summary

closes #651 

## Reviewer Checklist

> Reviewers should tick the following boxes before approving and merging the PR.

- [ ] The PR targets the `develop` branch
- [ ] The PR is assigned to the developer
- [ ] The PR has appropriate labels
- [ ] The PR is included in appropriate projects and/or milestones
- [ ] The PR includes a clear description of the proposed changes
- [ ] If the PR addresses an open issue the description includes "closes #<ISSUE NUMBER>"
- [ ] The code and documentation style match the current standards
- [ ] Documentation has been added/updated consistently with the code
- [ ] All CI tests are passing
- [ ] API docs have been built and checked at least once (if relevant)
- [ ] All changed files have been checked and comments provided to the developer
- [ ] All of the reviewer's comments have been satisfactorily addressed by the developer

### CODE REVIEW

File: install_shapepipe [hunk 1]
```diff
@@ -200,26 +202,28 @@ check_conda() {
     echo -ne "${RED}ERROR: Found Conda version $CONDA_VERSION but require 4.0.0 or greater.${NC}\n"
     exit 1
   fi
-  # Check Conda minor version
-  if [ "$CONDA_VERSION_MINOR" -gt 4 ]
+  # Look for conda.sh file
+  if [ -f "$CONDA_PREFIX_1$CONDA_SH" ]
```
Comment (martinkilbinger) [inline install_shapepipe:164]: $CONDA_PREFIX_1 does not seem to exist elsewhere in this script.

Comment (sfarrens) [reply install_shapepipe:164]: fixed!

**Comment by sfarrens** [comment]
Hi @martinkilbinger, after discussing with @cailmdaley we decided to move towards a Conda-free distribution of ShapePipe. Therefore, I think this PR can be merged as it is to have a stable installation with passing CI in the develop branch. Don't be scared by the number of changed files as this just because I have moved everything into a `src` directory to better fit the current community standard.

The default `environment.yml` has not been properly checked, but this will be dropped soon anyway. Anyone wishing to install ShapePipe in the meantime can just install the development environment.

```
./install_shapepipe --env-dev
```

If all looks good to you, we can merge and @cailmdaley can create a new branch from `develop`.

File: LICENSE
Comment (martinkilbinger) [inline LICENSE:1]: Why is there a French license file included?

File: environment-dev.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  - tqdm>=4.66
+  - treecorr>=4.3
+  - wget
+  - pip:
+    - cs_util>=0.0.5
+    - mccd>=1.2.3
+    - modopt>=1.6
+    - sip_tpv>=1.1
+    - sf_tools>=2.0.4
+    - git+https://github.com/CEA-COSMIC/pysap@develop
```
Comment (martinkilbinger) [inline environment-dev.yml]: Should we not use the pypi version? I tried with 0.2.1 and it seems to work.

**Review by martinkilbinger — APPROVED** [review]

File: environment-dev.yml (hunk 2 above)
Comment (sfarrens) [reply environment-dev.yml]: Done!

File: LICENSE
Comment (sfarrens) [reply LICENSE:1]: Oh weird, I just grabbed the first thing on Google. I didn't notice it was in French. Fixed!

**Review by martinkilbinger — APPROVED** [review]

**Pull request merged**